### DISCUSSION TITLE
Issue while updating document where auto-generated id's are getting generated again - V 2.2.1

### DISCUSSION THREAD

**Discussion by kunalfleethawks** [sensedeep/dynamodb-onetable#262] [Q&A] [answered]
Issue while updating document where auto-generated id's are getting generated again. 

Here is how our model looks

Library-Version: v2.2.1

```javascript
const table = new Table({
    name: "TestCrud",
    client: client,
    logger: true,
    uuid: () => {
        return ksuid.randomSync().string;
    },
    schema: {
        format: "onetable:1.1.0",
        version: "0.0.1",
        indexes: {
            primary: { hash: "pk", sort: "sk" },
            gs1: { hash: "gs1pk", sort: "gs1sk" },
        },
        models: {
            User: {
                pk: { type: String, required:true },
                sk: { type: String, value: "user#${id}" },
                id: { type: String, uuid:true, required:true },
                name: { type: String, required: true },
                status: { type: String, default: "active" },
                zip: { type: String },

                gs1pk: { type: String, value: "sec#${name}" },
                gs1sk: { type: String, value: "sec#${id}" },
            },
        },
        params: {
            isoDates: true,
            timestamps: true,
        },
    },
});
```

From the above code snippet, we are using UUID generation using ksuid (library) `id` in schema field is using UUID.

Insert works fine. But when we try to update the record id using pk and sk, the `id` field is updated with a new UUID (which is ksuid) in our case. 

Example:

Record in DB has following values before update
```
{
 pk: '12345',
 sk: 'user#249KEvpikVuDE4NoihiCG3S4Sqz',
 id:'249KEvpikVuDE4NoihiCG3S4Sqz',
 name: 'Jane'
 ......

}
```

After executing update statement: 

```
  user = await User.update({ pk:'12345', sk: 'user#249KEvpikVuDE4NoihiCG3S4Sqz', name:'Joe' });
    console.log("UPDATED user", user);
```
Record in DB
```
{
 pk: '12345',
 sk: 'user#249KEvpikVuDE4NoihiCG3S4Sqz',
 id:'204oAiLBpIDbEc8RRmlKhYIFNcw',
 name: 'Joe'
 ......

}
````

So if you look at `id` it is generated again. Ideally, it should be `249KEvpikVuDE4NoihiCG3S4Sqz`.

When we tried downgrading the library version to v1.8.0 it started working as expected. Can you suggest a workaround for this? 

thanks a lot.
Kunal

**Comment by mobsense** [comment, marked as answer]
We tried to reproduce your sample, but it wan't complete and our reproduction of it worked fine with 2.3.

Looking at the code, setDefaults will not return without generating IDs if update is called without params exists: null.  If you did not specify that, no IDs should be generated.

Can you please create a reproducible test case as per the issue template description under issues that we can run. Please make it as minimal as possible with no external deps.
